src/lib/logical_query_plan/predicate_node.hpp

```diff
@@ -38,10 +38,12 @@ class PredicateNode : public AbstractLQPNode {
       const std::shared_ptr<AbstractLQPNode>& right_child = nullptr) const override;
 
  protected:
-  std::shared_ptr<AbstractLQPNode> _deep_copy_impl() const override;
+  std::shared_ptr<AbstractLQPNode> _deep_copy_impl(
+      const std::shared_ptr<AbstractLQPNode>& copied_left_child,
+      const std::shared_ptr<AbstractLQPNode>& copied_right_child) const override;
 
  private:
-  const ColumnID _column_id;
+  const LQPColumnReference _column_references;
```

Comment:
Did you call it _column_reference**S** on purpose?

Reply:
nope